backend/api/Areas/Tools/Helpers/ImportPropertiesHelper.cs

```diff
@@ -0,0 +1,410 @@
+
+            if (agency == null)
+            {
+                agency = new Entity.Agency(property.Agency.Trim(), property.Agency);
+                _pimsAdminService.Agency.Add(agency);
+                _agencies.Add(agency);
+                _logger.LogDebug($"Adding agency '{agency.Code}' - '{agency.Name}'.");
+            }
+
+            if (!String.IsNullOrWhiteSpace(property.SubAgency))
+            {
+                var code = new string(property.SubAgency.GetFirstLetterOfEachWord(true).Take(6).ToArray());
+                var sub_agency = _agencies.FirstOrDefault(a => a.Code == code.Trim() && a.ParentId == agency.Id);
+
+                if (sub_agency == null)
+                {
+                    sub_agency = new Entity.Agency(code.Trim(), property.SubAgency)
+                    {
+                    ParentId = agency.Id,
+                    Parent = agency
+                    };
+                    _agencies.Add(sub_agency);
+                    _logger.LogDebug($"Adding sub-agency '{sub_agency.Code}' - '{agency.Name}', parent: '{sub_agency.Parent.Code}'.");
+                }
+
+                return sub_agency;
+            }
+
+            return agency;
+        }
+
+        /// <summary>
+        /// Get or create a new city for the specified code.
+        /// This is a recursive function so that it can generate a new city with a unique code.
+        /// It has a limitation of which the code must be 4 characters and therefore it can only handle 9 variations (i.e. BRe1, BRe2...)
+        /// </summary>
+        /// <param name="code"></param>
+        /// <param name="name"></param>
+        /// <param name="cities"></param>
+        /// <param name="attempts"></param>
+        /// <returns></returns>
+        private Entity.City GetOrCreateCity(string code, string name, int attempts = 1)
+        {
+            var city = _cities.FirstOrDefault(c => c.Code == code);
+
+            // If City doesn't exist, create it.
+            if (city == null)
+            {
+                city = new Entity.City(code, name);
+                _pimsAdminService.City.Add(city);
+                _cities.Add(city);
+
+                _logger.LogDebug($"Adding city '{code}' - '{name}'.");
+            }
+            else
+            {
+                // Generate a new city code.
+                code = $"{new string(code.Take(3).ToArray())}{attempts}";
+
+                // Check if the code is unique.
+                city = GetOrCreateCity(code, name, ++attempts);
+            }
+
+            return city;
+        }
+
+        /// <summary>
+        /// Get or create a new city for the specified name.
+        /// It will generate a unique city code based on the first initials of the city name, plus the first three characters of the last word.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="cities"></param>
```

Comment:
cities isn't a parameter.

Reply:
actually, just double check the param comments on this file, there are a few mismatches.

Reply:
Nice catch.